internal/server/data/data.go

```diff
@@ -59,14 +67,68 @@ type DB struct {
 	DefaultOrgSettings *models.Settings
 }
 
-func (db *DB) Close() error {
-	sqlDB, err := db.DB.DB()
+func (d *DB) Close() error {
+	sqlDB, err := d.DB.DB()
 	if err != nil {
 		return fmt.Errorf("failed to get database conn to close: %w", err)
 	}
 	return sqlDB.Close()
 }
 
+func (d *DB) DriverName() string {
+	return d.Dialector.Name()
+}
+
+func (d *DB) Exec(query string, args ...any) (sql.Result, error) {
+	db := d.DB.Exec(query, args...)
+	return driver.RowsAffected(db.RowsAffected), db.Error
+}
+
+func (d *DB) Query(query string, args ...any) (*sql.Rows, error) {
+	return d.DB.Raw(query, args...).Rows()
+}
+
+func (d *DB) QueryRow(query string, args ...any) *sql.Row {
+	return d.DB.Raw(query, args...).Row()
+}
+
+type GormDB interface {
```

Comment:
We probably don't need all of this yet. I'll see if I can reduce this interface a bit.